### PR TITLE
Scrub namespace registry redux

### LINKED ISSUE #1043: Scrub namespace registry config vs. csv to ensure it is an exact one-to-one mapping
### Are you sure this is not a new requirement or bug?

No

### Task Type

Sub-task

### 💡 Description

Per #1042 and https://github.com/NASA-PDS/pds4-information-model/pull/1041, there is clearly a disconnect here. Scrub the registry to make sure that:

1. everything in the csv is in the properties file
2. everything in the properties file is in the csv

### PR DESCRIPTION
**Pull request by jshughes**
Scrub namespace registry config vs. csv to ensure it is an exact one-to-one mapping #1043

## 🗒️ Summary
This redo of the namespace registry scrub consists of three commits.
1) Normalize pds-namespace-registry.csv file format
	Changes:
	force 7 columns and no trailing commas
	remove Line endings (CRLF/LF)
	cleanup Whitespace
	normalize CSV formatting
        Note:  All rows changed due to the change in the number of columns. ChatGPT was use to confirm that there were no content changes.

2) Update pds-namespace-registry.csv content
      Changes:
      New namespaces
      Steward changes
      Email corrections
      Record additions/deletions

3) Replace the original config.properties with the updated version
    Changes:
    Added insight, ladee, mgs, mpf, mvn and orex 

### 🤖 AI Assistance Disclosure
- [ ] No AI assistance used
- [x ] AI used for light assistance (e.g., suggestions, refactoring, documentation help, minor edits)
- [ ] AI used for moderate content generation (AI generated some code or logic, but the developer authored or heavily revised the majority)
- [ ] AI generated substantial portions of this code

Estimated % of code influenced by AI: _30__ %

## ⚙️ Test Data and/or Report
The manual tests include:
   Added six cucumber tests: github1043
   Used "diff" to validate that the list of namespaces in pds-namespace-registry.csv and config.properties were equivalent
   Extensive use of ChatGPT to identify formatting issues, formulate cleanup process, and validate results.
   Used "git diff" to visually validate commits.
  

## ♻️ Related Issues
Resolves #1043

## 🤓 Reviewer Checklist

*Reviewers: Please verify the following before approving this pull request.*

### Documentation and PR Content
- [ ] **Documentation:** README, Wiki, or inline documentation (Sphinx, Javadoc, Docstrings) have been updated to reflect these changes.
- [ ] **Issue Traceability:** The PR is linked to a valid GitHub Issue
- [ ] **PR Title:** The PR title is "user-friendly" clearly identifying what is being fixed or the new feature being added, that if you saw it in the Release Notes for a tool, you would be able to get the gist of what was done.

### Security & Quality
- [ ] **SonarCloud:** Confirmed no new High or Critical security findings.
- [ ] **Secrets Detection:** Verified that the Secrets Detection scan passed and no sensitive information (keys, tokens, PII) is exposed.
- [ ] **Code Quality:** Code follows organization style guidelines and best practices for the specific language (e.g., PEP 8, Google Java Style).

### Testing & Validation
- [ ] **Test Accuracy:** Verified that test data is accurate, representative of real-world PDS4 scenarios, and sufficient for the logic being tested.
- [ ] **Coverage:** Automated tests cover new logic and edge cases.
- [ ] **Local Verification:** (If applicable) Successfully built and ran the changes in a local or staging environment.

### Maintenance
- [ ] **Backward Compatibility:** Confirmed that these changes do not break existing downstream dependencies or API contracts (or that breaking changes are clearly documented).

### CODE REVIEW

File: model-ontology/src/ontology/Data/config.properties [hunk 1]
```diff
@@ -700,6 +755,17 @@ lSchemaFileDefn.ody.urnPrefix=urn:nasa:pds:
 lSchemaFileDefn.ody.modelShortName=PDS4
 lSchemaFileDefn.ody.regAuthId=0001_NASA_PDS_1
 
+lSchemaFileDefn.orex.identifier=orex
+lSchemaFileDefn.orex.isMaster=false
+lSchemaFileDefn.orex.isDiscipline=false
+lSchemaFileDefn.orex.stewardArr=psi
```
Comment (nutjob4life) [inline model-ontology/src/ontology/Data/config.properties:761]: It looks like for `orex`, the `stewardArr=psi`, but I don't see any `psi` in `docs/namespace-registry/pds-namespace-registry.csv`. Is that okay? (There is `orex`, though!)

File: model-ontology/src/ontology/Data/config.properties [hunk 2]
```diff
@@ -640,6 +673,17 @@ lSchemaFileDefn.mmx.modelShortName=DARTS
 lSchemaFileDefn.mmx.regAuthId=0001_JAXA_DARTS_1
 lSchemaFileDefn.mmx.comment=Namespace for the Martian Moons eXploration dictionary.
 
+lSchemaFileDefn.mpf.identifier=mpf
+lSchemaFileDefn.mpf.isMaster=false
+lSchemaFileDefn.mpf.isDiscipline=false
+lSchemaFileDefn.mpf.stewardArr=img
```
Comment (nutjob4life) [inline model-ontology/src/ontology/Data/config.properties:679]: Similar to above: for `mpf`, the `stewardArr` should be `img`? 

Not `mpf`? There are both `mpf` and `img` in `docs/namespace-registry/pds-namespace-registry.csv`

File: model-ontology/src/ontology/Data/config.properties [hunk 3]
```diff
@@ -619,15 +651,16 @@ lSchemaFileDefn.mgn.urnPrefix=urn:nasa:pds:
...
-lSchemaFileDefn.mer.stewardArr=geo
-lSchemaFileDefn.mer.nameSpaceURL=http://pds.nasa.gov/pds4/
-lSchemaFileDefn.mer.nameSpaceURLs=https://pds.nasa.gov/pds4/
-lSchemaFileDefn.mer.urnPrefix=urn:nasa:pds:
-lSchemaFileDefn.mer.modelShortName=PDS4
-lSchemaFileDefn.mer.regAuthId=0001_NASA_PDS_1
+lSchemaFileDefn.mgs.identifier=mgs
+lSchemaFileDefn.mgs.isMaster=false
+lSchemaFileDefn.mgs.isDiscipline=false
+lSchemaFileDefn.mgs.stewardArr=img
```
Comment (nutjob4life) [inline model-ontology/src/ontology/Data/config.properties:657]: Double-check: for `mgs`, the `stewardArr` should be `img`? 

Not `mgs`? There are both `mgs` and `img` in `docs/namespace-registry/pds-namespace-registry.csv`

File: docs/namespace-registry/pds-namespace-registry.csv [hunk 4]
```diff
@@ -1,167 +1,148 @@
...
+clps,clps,"The Commercial Lunar Payload Services Mission Dictionary (clps) contains classes, attributes and rules specific to the CLPS missions and their instruments.",Mission,Jennifer Ward,jgward at wustl.edu,2023-03-28
+cots_camera,COTS Camera,"The COTS Camera Mission Dictionary (cots_camera) contains classes, attributes and rules specific to commercial off-the-shelf camera instruments.",Mission,Bob Deen,Bob.Deen at jpl.nasa.gov,2026-04-15
+dart,dart,"This namespace provides classes specific to the NASA DART mission and, potentially, the concurrent LICIACube mission.",Mission,Ben Hirsch,bhirsch1 at umd.edu,2021-08-18
+dscovr,DSCOVR Mission,The DSCOVR Mission Data Dictionary contains classes and attributes that describe data products from the DSCOVR (Deep Space Climate Observatory) mission.,Mission,Bob Deen,Bob.Deen at jpl.nasa.gov,2025-07-30
+galileo,Galileo,"The Galileo Mission Dictionary (galileo) contains classes, attributes and rules specific to the Galileo Mission and its instruments.",Mission,Jesse Stone; Beatrice Mueller,jstone at psi.edu; mueller at psi.edu,2025-09-23
```
Comment (nutjob4life) [inline docs/namespace-registry/pds-namespace-registry.csv:55]: The PR says "ChatGPT was use to confirm that there were no content changes" but this line seems to have changed content.

Previously, it had 5 columns and was followed by 2 rows of 2 columns. Now those rows are gone and have joined line 55 to become 7 columns. Just want to double-check that that's okay!

File: docs/namespace-registry/pds-namespace-registry.csv [hunk 5]
```diff
@@ -1,167 +1,148 @@
...
-,,,,,,,
-(1) Namespace Id is defined in the PDS4 Information Model. It is a namespace container for a logical grouping of classes and attributes and is assigned by the steward. Namespace_Id is often mapped to the namespace prefix defined in XML documents.,,,,,,,
-"(2) The Namespace Id used to create a URL for a mission level dictionary requires the prefix ""mission"".",,,,,,,
-"(3) The default namespace and namespace prefix, in an XML Schema file, is null.",,,,,,,
-"(4) The Schema File Name Prefix typically has a suffix that includes the version number of the dictionary, for example PDS4_PDS_1400.",,,,,,,
-(5) The PDS Change Control Board (CCB) provides oversight by reviewing and approving all changes to the Common dictionary.,,,,,,,
-,,,,,,,
-,,,,,,,
-,,,,,,,
+Last Update,2026-06-16,,,,,
```
Comment (nutjob4life) [inline docs/namespace-registry/pds-namespace-registry.csv:1]: Was there a byte-order mark (BOM) at the top of this file? Was it removed? If so, is that okay?

**Review by nutjob4life — CHANGES_REQUESTED** [review]
Hi @jshughes! This is great! Thanks for putting all this together. I did have a few questions (see them interspersed with the code deltas) which might indicate some important bits—but I'm also highly unfamiliar with this, so they may be fine! 😆 Let me know what you think. Maybe they can all be resolved with just explanations and without any changes 🤞

File: model-ontology/src/ontology/Data/config.properties (hunk 1 above)
Comment (c-suh) [reply model-ontology/src/ontology/Data/config.properties:761]: Ooh, good point, nutjob4life. Similarly...
- `lSchemaFileDefn.ops.stewardArr=en` ([line 251](https://github.com/NASA-PDS/pds4-information-model/blob/1c97633f7ef56949bfa4bc6433294a4c4e964ef8/model-ontology/src/ontology/Data/config.properties#L251)) but no `en` in the namespace-registry
- these all say `sbn`. Is it supposed to be `sb` (for 'Small Bodies Node Dictionary')?
   - `lSchemaFileDefn.sb.stewardArr=sbn` [line 293](https://github.com/NASA-PDS/pds4-information-model/blob/1c97633f7ef56949bfa4bc6433294a4c4e964ef8/model-ontology/src/ontology/Data/config.properties#L293)
   - `lSchemaFileDefn.iras.stewardArr=sbn` [line 566](https://github.com/NASA-PDS/pds4-information-model/blob/1c97633f7ef56949bfa4bc6433294a4c4e964ef8/model-ontology/src/ontology/Data/config.properties#L566)
   - `lSchemaFileDefn.lucy.stewardArr=sbn` [line 617](https://github.com/NASA-PDS/pds4-information-model/blob/1c97633f7ef56949bfa4bc6433294a4c4e964ef8/model-ontology/src/ontology/Data/config.properties#L617)
   - `lSchemaFileDefn.near.stewardArr=sbn` [line 721](https://github.com/NASA-PDS/pds4-information-model/blob/1c97633f7ef56949bfa4bc6433294a4c4e964ef8/model-ontology/src/ontology/Data/config.properties#L721)
   - `lSchemaFileDefn.neas.stewardArr=sbn` [line 731](https://github.com/NASA-PDS/pds4-information-model/blob/1c97633f7ef56949bfa4bc6433294a4c4e964ef8/model-ontology/src/ontology/Data/config.properties#L731)
   - `lSchemaFileDefn.nh.stewardArr=sbn` [line 741](https://github.com/NASA-PDS/pds4-information-model/blob/1c97633f7ef56949bfa4bc6433294a4c4e964ef8/model-ontology/src/ontology/Data/config.properties#L741)
   - `lSchemaFileDefn.bopps.stewardArr=sbn` [line 887](https://github.com/NASA-PDS/pds4-information-model/blob/1c97633f7ef56949bfa4bc6433294a4c4e964ef8/model-ontology/src/ontology/Data/config.properties#L887)

File: docs/namespace-registry/pds-namespace-registry.csv (hunk 4 above)
Comment (c-suh) [reply docs/namespace-registry/pds-namespace-registry.csv:55]: I believe this was supposed to be one row (e.g., Jesse Stone and Beatrice Meller are both stewards for Galileo), so this was a correction. But yes, @jshughes, confirm please.

**Review by c-suh — COMMENTED** [review]
@jshughes what should we be looking for with the test files?

File: model-ontology/src/ontology/Data/config.properties (hunk 1 above)
Comment (jshughes) [reply model-ontology/src/ontology/Data/config.properties:761]: These and the following are all good calls but unfortunately, the DDWG needs to be involved to resolve the issues in almost all of the cases. I would like to suggest that we go ahead and merge the PR and use the change request, "Namespace Registry and Config Cleanup - Phase 2 #1052" to make any changes resulting from DDWG deliberation. I have included a note in each case to help explain the issue.

The main issue is, for a mission the steward id is typically the namespace id of the mission. However, this rule has never been formally decided by the DDWG.

So, for the "orex" mission, the steward_id would typically be "orex". I would like to get the DDWG to formally approve this rule.

For the "en" case, the engineering node has yet to be added to the stewards list. The DDWG needs to resolve this.

In the "sbn" case, the available steward_id in the IM is "sbn". The "sb" namespace came later. The DDWG or at least the SBN needs to resolve this.

File: model-ontology/src/ontology/Data/config.properties (hunk 2 above)
Comment (jshughes) [reply model-ontology/src/ontology/Data/config.properties:679]: The steward_id would typically be "mpf". I would like the DDWG to confirm this.

File: model-ontology/src/ontology/Data/config.properties (hunk 3 above)
Comment (jshughes) [reply model-ontology/src/ontology/Data/config.properties:657]: The steward_id would typically be "mgs". I would like the DDWG to confirm the rule.

File: docs/namespace-registry/pds-namespace-registry.csv (hunk 4 above)
Comment (jshughes) [reply docs/namespace-registry/pds-namespace-registry.csv:55]: Yes, Catherine is correct, the steward name and email address fields each had two values. The two values were originally separated by commas. The commas were replaced by semicolon to allow two values in each field.

File: docs/namespace-registry/pds-namespace-registry.csv (hunk 5 above)
Comment (jshughes) [reply docs/namespace-registry/pds-namespace-registry.csv:1]: Yes, there was a byte-order mark (BOM) and it was removed. Given that the file is maintained in GitHub, I understand that not having the BOM mark is generally the most robust choice. Please feel free to advise otherwise.

File: model-ontology/src/ontology/Data/config.properties (hunk 1 above)
Comment (nutjob4life) [reply model-ontology/src/ontology/Data/config.properties:761]: @jshughes wow, I learned a lot! Thanks for clarifying all this. I'm fine with merging and then following up with the DDWG. I'll change my disposition to "Approved".

**Review by nutjob4life — APPROVED** [review]
Now approved based on @jshughes explanations.

File: model-ontology/src/ontology/Data/config.properties (hunk 2 above)
Comment (jordanpadams) [reply model-ontology/src/ontology/Data/config.properties:679]: @jshughes hmmm. does the DDWG really need to be involved here? this is a DD Stewards thing not a DDWG thing. Either way, I don't think either have any idea what this stuff is. Usually a node is responsible for this LDDs, not the mission themselves.

File: docs/namespace-registry/pds-namespace-registry.csv [hunk 6]
```diff
@@ -1,167 +1,148 @@
...
+Namespace Id,Name,Description,Governance Level,Steward (Contact),Contact Email Address,Registration Date
+Common,,,,,,
+pds,Planetary Data System,Namespace for the PDS's common dictionary.,Common,Steve Hughes,Steve.Hughes at jpl.nasa.gov,2012-04-03
```
Comment (jordanpadams) [inline docs/namespace-registry/pds-namespace-registry.csv:4]: @jshughes I am again concerned that it is impossible to track what changed in this CSV because of some change that occurred to every line in this CSV. I am guessing this is a whitespace or newline change (CRLF vs. LF change) that may have been introduced if this was opened in Excel vs. maybe it was not in the past? Not sure.

I trust you have done your due diligence and double/triple checked we did not introduce a bug here. With the PR approved, I will leave it to you to click the green button to Merge this pull request ensuring that you are confident in this merge.

If there is any way we in the future we can avoid this, that would be ideal.

**Review by jordanpadams — APPROVED** [review]

File: docs/namespace-registry/pds-namespace-registry.csv (hunk 6 above)
Comment (jshughes) [reply docs/namespace-registry/pds-namespace-registry.csv:4]: @jordanpadams @nutjob4life @c-suh Got it. As you know, the root cause was that the first commit intentionally normalized the CSV formatting which caused Git to show every row as modified. (The two key problems were CRLF instead of LF and an extract comma on each line.) Given that this was a one-time update, future pull requests should contain only content changes, making the Git diffs small and easy to review.

As an additional safeguard, I prompted Claude-code to write a program to validate using the new “canonical” CSV format. It will be used as a validation step as part of future updates. It just found a trailing blank in one field. It was removed in the last commit to this PR.

CSV format validation: C:\Users\jshughes\claude-code\NSCSVValidateFormat\pds-namespace-registry.csv

[PASS] UTF-8 encoding
        File decodes cleanly as UTF-8.
[PASS] No UTF-8 byte-order mark (BOM)
        No BOM present.
[PASS] LF line endings
        148 line(s) all terminated with LF.
[PASS] Column count == 7
        Every row has 7 columns.
[PASS] No leading/trailing field whitespace
        No field has leading or trailing whitespace.

Result: VALID - all format rules passed.

Exit code: 0

Comment (nutjob4life) [reply docs/namespace-registry/pds-namespace-registry.csv:4]: @jshughes okie doke, thanks!

**Comment by jordanpadams** [comment]
thanks for checking @jshughes ! much appreciated. apologies for being so pedantic!

**Comment by jshughes** [comment]
@jordanpadams No problem and I always appreciate suggestions for more robust solutions..
